yunwoo/baekjoon/문자열/15829-Hashing.js

```diff
@@ -0,0 +1,61 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const l = +input[0];
+  const englishString = input[1];
+  const m = 1234567891;
+  const alpabet = {};
+  let r = 1;
+  let result = 0;
+
+  Array.from(
+    { length: 26 },
+    (_, i) => (alpabet[String.fromCharCode(i + 97)] = i + 1)
+  );
+
+  for (let i = 0; i < l; i++) {
+    result += (alpabet[englishString[i]] * r) % m;
+    r = (r * 31) % m;
+  }
+
+  console.log(result % m);
+});
```

Comment:
```suggestion
  const l = +input[0];
  const englishString = input[1];
  const m = 1234567891;
  let r = 1;
  let result = 0;

  for (let i = 0; i < l; i++) {
    result += (englishString.charCodeAt(i) - 96) * r;
    r = (r * 31) % m;
  }

  console.log(result % m);
```
- 풀이 잘봤습니다! 🙇‍♂️
- ASCII 코드 변환을 이용한 풀이를 이용했다면 charCodeAt을 활용하는 것도 좋습니다 :) 
- 그리고 result는 매번 1234567891로 나눠서 나머지를 할당 할 필요없이 맨 마지막 출력 시에 1번만 진행하셔도 됩니다 🙇‍♂️